### PR TITLE
Apply package.json overrides after frozen install

### PR DESCRIPTION
**Pull request by masenf**
### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Description

Fixes the issue where upgrading Reflex to a version that introduces a new `package.json` override fails with `lockfile had changes, but lockfile is frozen`.

**Root cause:** Framework overrides were being merged into `package.json` while restoring the persisted file from `reflex.lock/`, which would desync it from the persisted lockfile and cause `--frozen-lockfile` to fail.

**Solution:** Defer merging framework overrides until after the frozen install completes. This keeps the restored `package.json`/lockfile pair intact for the frozen install, then applies overrides via a new `update_package_json_overrides()` function before any dependency resolution happens.

### Changes

**Core logic (`reflex/utils/frontend_skeleton.py`):**
- Refactored `_read_persisted_package_json()` into a generic `_read_package_json_object()` helper
- Modified `_compile_package_json()` to preserve persisted overrides verbatim (no framework merging)
- Added `update_package_json_overrides()` to merge framework overrides into `.web/package.json` after the frozen install
- Updated docstrings to clarify the new override handling strategy

**Install flow (`reflex/utils/js_runtimes.py`):**
- Call `update_package_json_overrides()` after the frozen install
- If overrides changed but no packages are being added, run a non-frozen install to resolve the new overrides into the lockfile
- Updated cache payload to include overrides so changes bust the install cache

**Tests (`tests/units/test_prerequisites.py`):**
- Added `_stub_framework_packages()` to also stub `OVERRIDES` for predictable test counts
- Added 4 new integration tests covering:
  - Applying new overrides after frozen install while preserving user overrides
  - Reconciling lockfile when only overrides change
  - Skipping reconcile when overrides are already applied
  - Cache invalidation when overrides change
- Added 5 new unit tests for `update_package_json_overrides()` covering merging, missing sections, no-ops, and malformed files
- Updated existing tests to reflect that framework overrides are no longer merged during compilation

### Test Plan

All new tests pass and cover the override handling logic:
- `test_install_frontend_packages_applies_overrides_after_frozen_install` validates the core fix
- `test_install_frontend_packages_reconciles_lockfile_for_overrides_only_change` ensures lockfile is refreshed when needed
- `test_install_frontend_packages_skips_reconcile_when_overrides_unchanged` prevents unnecessary work
- `test_install_frontend_packages_cache_invalidated_by_new_override` ensures cache busting works
- Unit tests for `update_package_json_overrides()` cover all edge cases

Existing tests updated to match new behavior where framework overrides are applied post-frozen-install rather than during compilation.

closes #6792

https://claude.ai/code/session_018WXodmwe6tFZ6DkweqArij

<a href="https://cubic.dev/pr/reflex-dev/reflex/pull/6844?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: reflex/utils/js_runtimes.py [hunk 1]
```diff
@@ -754,6 +760,14 @@ def _install_frontend_packages(
             show_status_message="Installing frontend packages",
         )
 
+    if overrides_changed and not (dev_deps_to_add or deps_to_add):
```
[the remark this answers is not in the document]
Comment (masenf) [reply reflex/utils/js_runtimes.py:763]: going to keep it for now. it's not causing harm and in the future if the install flow changes, we need to guard against this edge case that doesn't come up very often (we rarely change overrides)

File: reflex/utils/frontend_skeleton.py [hunk 2]
```diff
@@ -542,14 +556,47 @@ def _compile_package_json():
...
+        Whether ``.web/package.json`` was changed.
+    """
+    package_json_path = get_web_lockfile_path(constants.PackageJson.PATH)
+    package_json = _read_package_json_object(package_json_path)
+    if not package_json:
+        # Nothing usable to merge into. initialize_package_json() renders a
+        # full file upstream, so this only happens for a hand-damaged .web.
+        return False
+
+    overrides = package_json.get("overrides") or {}
```
[the remark this answers is not in the document]
Comment (masenf) [reply reflex/utils/frontend_skeleton.py:587]: if the user comes with a malformed package.json, that's on them, we're not going to change/break what they provide. if the app doesn't start, then the user can fix or remove their locked package.json

**Review by FarhanAliRaza — APPROVED** [review]
